cpp/src/replace/replace.cu

```diff
@@ -0,0 +1,209 @@
+#include <bitmask/legacy/legacy_bitmask.hpp>
+#include <bitmask/legacy/bit_mask.cuh>
+
+namespace {  // anonymous
+
+using namespace cudf;
+
+static constexpr int BLOCK_SIZE = 256;
+
+/* --------------------------------------------------------------------------*/
+/**
+ * @brief Kernel that converts inputs from `in` to `out`  using the following
+ *        rule:   Convert  -NaN  -> NaN
+ *                Convert  -0.0  -> 0.0
+ *
+ * @param[in] column_device_view representing input data
+ * @param[in] mutable_column_device_view representing output data. can be
+ *            the same actual underlying buffer that in points to. 
+ *
+ * @returns
+ */
+/* ----------------------------------------------------------------------------*/
+template <typename T>
+__global__
+void normalize_nans_and_zeros(column_device_view in, 
+                              mutable_column_device_view out)
+{
+   int tid = threadIdx.x;
+   int blkid = blockIdx.x;
+   int blksz = blockDim.x;
+   int gridsz = gridDim.x;
+
+   int start = tid + blkid * blksz;
+   int step = blksz * gridsz;
+
+   // grid-stride
+   for (int i=start; i<in.size(); i+=step) {
+      if(!in.is_valid(i)){
+         continue;
+      }
+
+      T el = in.element<T>(i);
+      if(std::isnan(el)){
+         out.element<T>(i) = std::numeric_limits<T>::quiet_NaN();
+      } else if(el == (T)-0.0){
+         out.element<T>(i) = (T)0.0;
+      } else {
+         out.element<T>(i) = el;
+      }
+   }
+}                        
+
+  /* --------------------------------------------------------------------------*/
+  /**
+   * @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+   *        `normalize_nans_and_zeros` with the appropriate data types.
+   */
+  /* ----------------------------------------------------------------------------*/
+struct normalize_nans_and_zeros_kernel_forwarder {
+   // floats and doubles. what we really care about.
+   template <typename T, std::enable_if_t<std::is_floating_point<T>::value>* = nullptr>
+   void operator()(  column_device_view in,
+                     mutable_column_device_view out,
+                     cudaStream_t stream)
+   {
+      util::cuda::grid_config_1d grid{in.size(), BLOCK_SIZE};
+      normalize_nans_and_zeros<T><<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(in, out);
+   }
+
+   // if we get in here for anything but a float or double, that's a problem.
+   template <typename T, std::enable_if_t<not std::is_floating_point<T>::value>* = nullptr>
+   void operator()(  column_device_view in,
+                     mutable_column_device_view out,
+                     cudaStream_t stream)
+   {
+      CUDF_FAIL("Unexpected non floating-point type.");      
+   }   
+};
+
+} // end anonymous namespace
+
+namespace cudf {
+namespace detail {
+
+std::unique_ptr<column> normalize_nans_and_zeros( column_view input,                                            
+                                                  cudaStream_t stream,
```

Comment:
```suggestion
                                                  cudaStream_t stream = 0,
```

Reply:
Fixed in ddd408a26f3ada45a156fe5c9df6d8a3746dbd23   (this function goes away)